### PR TITLE
Project management: Add prepublish packages command for npm releases

### PR DESCRIPTION
**Pull request by gziolo**
## Description
This PR adds a new command which handles prepublish to npm steps for a stable version of WordPress packages.

At the moment it only covers those steps which handle git management before Lerna takes over publishing to npm.

Check more details about the manual process in https://github.com/WordPress/gutenberg/blob/master/docs/contributors/release.md#synchronizing-wordpress-trunk.

## How has this been tested?
`./bin/commander.js prepublish-packages-stable`

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR. .

### CODE REVIEW

File: bin/commander.js [hunk 1]
```diff
@@ -672,6 +685,83 @@ program
...
+
+	// Checking out the WordPres release branch and doing sync with the last plugin relase.
+	const { releaseBranch } = await runWordPressReleaseBranchSyncStep( abortMessage );
+
+	// Push the local changes
+	abortMessage = `Aborting! Make sure to push changes applied to WordPress release branch "${ releaseBranch }" manually.`;
+	await runPushGitChangesStep( releaseBranch, abortMessage );
+
+	abortMessage = 'Aborting! The release is finished though.';
+	await runCleanLocalCloneStep( abortMessage );
```
Comment (gziolo) [inline bin/commander.js]: This references svn folder which doesn’t exist in this context causing errors.

File: bin/commander.js [hunk 2]
```diff
@@ -672,6 +685,83 @@ program
...
+ */
+async function prepublishPackages() {
+	// This is a variable that contains the abort message shown when the script is aborted.
+	let abortMessage = 'Aborting!';
+	await askForConfirmationToContinue( 'Ready to go? ' );
+
+	// Cloning the Git repository.
+	await runGitRepositoryCloneStep( abortMessage );
+
+	// Checking out the WordPres release branch and doing sync with the last plugin relase.
```
Comment (youknowriad) [inline bin/commander.js]:
```suggestion
	// Checking out the WordPress release branch and doing sync with the last plugin release.
```

**Review by youknowriad — APPROVED** [review]
I haven't tested, but it's looking good.

Good job.

File: bin/commander.js [hunk 3]
```diff
@@ -237,8 +237,18 @@ async function updateThePluginStableVersion( version, abortMessage ) {
...
 async function runCleanLocalCloneStep( abortMessage ) {
 	await runStep( 'Cleaning the temporary folder', abortMessage, async () => {
-		await rimraf( gitWorkingDirectoryPath );
-		await rimraf( svnWorkingDirectoryPath );
+		await Promise.all( [
+			gitWorkingDirectoryPath,
+			svnWorkingDirectoryPath,
+		].map( async ( directoryPath ) => {
+			if ( fs.existsSync( directoryPath ) ) {
+				await rimraf( directoryPath, ( err ) => {
```
Comment (gziolo) [inline bin/commander.js]: They do the same trick in `rimraf` in their bin implementation: 
https://github.com/isaacs/rimraf/blob/d82bc81f251ba2cc86dc26361a820631091b3e9e/bin.js#L44-L48

**Comment by gziolo** [comment]
It was tested already on `wp/trunk` branch and everything worked properly, see:

https://github.com/WordPress/gutenberg/commit/94a0c81d1a0694af03e9262fcabf4601317f41ed

File: bin/commander.js (hunk 3 above)
Comment (gziolo) [reply bin/commander.js]: It fixes the following error I observed locally:

```
The following error happened during the "Cleaning the temporary folder" step:
 AssertionError [ERR_ASSERTION]: rimraf: callback function required
    at rimraf (/Users/gziolo/PhpstormProjects/gutenberg/node_modules/rimraf/rimraf.js:52:10)
```

**Pull request merged**